tests/testutils/telemetry/metrics.go

```diff
@@ -256,13 +256,23 @@ func (metric Metric) equals(toCompare Metric, strict bool) bool {
 		return false
 	}
 
-	if metric.Value != toCompare.Value && (strict || metric.Value != nil) {
+	if !valuesEqual(metric.Value, toCompare.Value) && (strict || metric.Value != nil) {
```

Comment:
I think generics would be faster but testify provides a reflection based helper that would account for float64 vals too

```suggestion
	if !assert.ObjectsAreEqualValues(metric.Value, toCompare.Value) && (strict || metric.Value != nil) {
```

Reply:
great, thanks, applying